scripts/tools/combine-tools.ts

```diff
@@ -192,13 +191,42 @@ const combineTools = async (
             (await Promise.all(manualTools[key].toolsList.map(processManualTool))).filter(Boolean)
           : [];
 
-        finalTools[key].toolsList = [...automatedResults, ...manualResults].sort((tool, anotherTool) =>
-          tool!.title.localeCompare(anotherTool!.title)
-        ) as FinalAsyncAPITool[];
+        finalTools[key].toolsList = [...automatedResults, ...manualResults].sort((tool, anotherTool) => {
+          if (!tool?.title || !anotherTool?.title) {
+            logger.error({
+              message: 'Tool title is missing during sort',
+              detail: { tool, anotherTool },
+              source: 'combine-tools.ts'
+            });
+
+            return 0;
+          }
+
+          return tool.title.localeCompare(anotherTool.title);
+        }) as FinalAsyncAPITool[];
       }
     }
-    fs.writeFileSync(toolsPath, JSON.stringify(finalTools));
-    fs.writeFileSync(tagsPath, JSON.stringify({ languages: languageList, technologies: technologyList }));
+    try {
+      fs.writeFileSync(toolsPath, JSON.stringify(finalTools, null, 2));
+      fs.writeFileSync(
+        tagsPath,
+        JSON.stringify(
+          {
+            languages: languageList,
+            technologies: technologyList
+          },
+          null,
+          2
+        )
+      );
+    } catch (error) {
+      throw new Error(
+        'Failed to write output files:\n' +
+          `Tools path: ${toolsPath}\n` +
+          `Tags path: ${tagsPath}\n` +
+          `Error: ${(error as Error).message}`
+      );
+    }
```

Comment:
Why nested try/catch has been added?

Reply:
I didn't realize I was adding nested try catches, I have added this catch for better debugging purposes. it prints the whole tools and tags path in the error message. Let me remove it.